### PR TITLE
send multiple event batches in one replication task

### PR DESCRIPTION
**Pull request by hai719**
## What changed?
Only one replication task is generated even if there are multiple event batches in one transaction on the active side.

## Why?
We need make sure events that belong to the same transaction on the active side also belong to the same transaction on the standby side.

## How did you test it?
Tested locally. Added new tests.

### CODE REVIEW

File: proto/internal/temporal/server/api/replication/v1/message.proto [hunk 1]
```diff
@@ -169,6 +169,7 @@ message HistoryTaskAttributes {
     temporal.api.common.v1.DataBlob new_run_events = 7;
     temporal.server.api.workflow.v1.BaseExecutionInfo base_execution_info = 8;
     string new_run_id = 9;
+    repeated temporal.api.common.v1.DataBlob events_batches = 10;
```
Comment (xwduan) [inline proto/internal/temporal/server/api/replication/v1/message.proto:172]: Maybe mark line 167 as deprecated?

File: service/history/workflow/task_generator.go [hunk 2]
```diff
@@ -694,14 +694,16 @@ func (r *TaskGeneratorImpl) GenerateUserTimerTasks() error {
...
+	if len(eventBatches) == 0 {
 		return nil
 	}
 
-	firstEvent := events[0]
-	lastEvent := events[len(events)-1]
+	firstBatch := eventBatches[0]
+	firstEvent := firstBatch[0]
+	lastBatch := eventBatches[len(eventBatches)-1]
+	lastEvent := lastBatch[len(lastBatch)-1]
```
Comment (yux0) [inline service/history/workflow/task_generator.go:706]: I think we need the first event version abd last event version to locate which history branch to use

Comment (yux0) [reply service/history/workflow/task_generator.go:706]: Or at least the version should be last event version to locate the history branch.

Comment (yux0) [reply service/history/workflow/task_generator.go:706]: Synced offline. They will be generated from one transaction. Could we add a sanity check to make sure they use the same version?

Comment (hai719) [reply service/history/workflow/task_generator.go:712]: check added.

File: proto/internal/temporal/server/api/replication/v1/message.proto (hunk 1 above)
Comment (hai719) [reply proto/internal/temporal/server/api/replication/v1/message.proto:174]: added.

File: service/history/workflow/task_generator.go [hunk 3]
```diff
@@ -694,14 +694,16 @@ func (r *TaskGeneratorImpl) GenerateUserTimerTasks() error {
 }
 
 func (r *TaskGeneratorImpl) GenerateHistoryReplicationTasks(
-	events []*historypb.HistoryEvent,
+	eventBatches [][]*historypb.HistoryEvent,
 ) error {
-	if len(events) == 0 {
+	if len(eventBatches) == 0 {
```
[the remark this answers is not in the document]
Comment (hai719) [reply service/history/workflow/task_generator.go:700]: check added.

**Review by yux0 — APPROVED** [review]

**Review by xwduan — APPROVED** [review]

**Pull request merged**